packages/ilios-common/addon/components/course/objectives.js

```diff
@@ -21,13 +21,18 @@ export default class CourseObjectivesComponent extends Component {
     return this.objectiveCount > 0;
   }
 
-  get objectiveCount() {
-    return this.objectivesRelationship ? this.objectivesRelationship.length : 0;
+  @cached
+  get objectivesData() {
+    return new TrackedAsyncData(this.args.course.courseObjectives);
   }
 
-  load = restartableTask(async () => {
-    this.objectivesRelationship = await this.args.course.courseObjectives;
-  });
+  get objectives() {
+    return this.objectivesData.isResolved ? this.objectivesData.value : null;
```

Comment:
similar comment as above - return an empty array here if `objectivesData` is unresolved.

```suggestion
    return this.objectivesData.isResolved ? this.objectivesData.value : [];
```    

then you can also throw out the `hasObjectives` and `objectivesCount` getters and use `this.objectives.length` directly in their place where needed.